lib-rpc/ocamlformat_rpc_lib.mli

```diff
@@ -9,70 +9,75 @@
 (*                                                                        *)
 (**************************************************************************)
 
-module type Command_S = sig
-  type t
+module Make (IO : IO.S) : sig
+  module type Command_S = sig
+    type t
 
-  val read_input : Stdlib.in_channel -> t
+    val read_input : IO.ic -> t IO.t
 
-  val to_sexp : t -> Sexplib0.Sexp.t
+    val to_sexp : t -> Csexp.t
 
-  val output : Stdlib.out_channel -> t -> unit
-end
+    val output : IO.oc -> t -> unit IO.t
+  end
 
-module type Client_S = sig
-  type t
+  module type Client_S = sig
+    type t
 
-  type cmd
+    type cmd
 
-  val pid : t -> int
+    val pid : t -> int
 
-  val mk : pid:int -> in_channel -> out_channel -> t
+    val mk : pid:int -> IO.ic -> IO.oc -> t
 
-  val query : cmd -> t -> cmd
+    val query : cmd -> t -> cmd IO.t
 
-  val halt : t -> (unit, [> `Msg of string]) result
+    val halt : t -> (unit, [> `Msg of string]) result IO.t
 
-  val config :
-    (string * string) list -> t -> (unit, [> `Msg of string]) result
+    val config :
+      (string * string) list -> t -> (unit, [> `Msg of string]) result IO.t
 
-  val format : string -> t -> (string, [> `Msg of string]) result
-end
+    val format : string -> t -> (string, [> `Msg of string]) result IO.t
+  end
 
-module type V = sig
-  module Command : Command_S
+  module type V = sig
+    module Command : Command_S
 
-  module Client : Client_S with type cmd = Command.t
-end
+    module Client : Client_S with type cmd = Command.t
+  end
 
-(** Version used to set the protocol version *)
-module Init : Command_S with type t = [`Halt | `Unknown | `Version of string]
+  (** Version used to set the protocol version *)
+  module Init :
+    Command_S with type t = [`Halt | `Unknown | `Version of string]
 
-module V1 :
-  V
-    with type Command.t =
-      [ `Halt
-      | `Unknown
-      | `Error of string
-      | `Config of (string * string) list
-      | `Format of string ]
+  module V1 :
+    V
+      with type Command.t =
+        [ `Halt
+        | `Unknown
+        | `Error of string
+        | `Config of (string * string) list
+        | `Format of string ]
```

Comment:
Why expose the commands ? It's not used in the API below.

Reply:
If we remove this we can't match on the command sent/received on the server side, maybe we could redesign the whole thing in a future PR